browser/src/Editor/NeovimEditor.tsx

```diff
@@ -283,6 +283,30 @@ export class NeovimEditor implements IEditor {
             }
         })
 
+        const normalizePath = (fileName: string) => fileName.split("\\").join("/")
+
+        ipcRenderer.on("open-files", (_evt: any, message: string, files: string[]) => {
+
+            // Open the first file.
+            // If the current buffer is named, then open in a new tab or split.
+            // If the current buffer has modifications pending, then open in a new tab or split.
+            // If the current buffer had no modification and is a new file, open in the current buffer.
+
+            this._neovimInstance.command("if bufname('%') != ''\n" +
```

Comment:
If we need extensive logic, it might be worth creating a function for this in `vim\core\oni-core-interop\plugin\init.vim`. Something like:

```
function OniOpenFile(strategy, file)
    if bufname('%') !== ''
        exec a:strategy . a:file
    elseif &modified
        exec a:strategy . a:file
    else
        exec ":e " . a:file
    endif
endfunction
```

You can then call this function by executing:
this._neovimInstance.callFunction("OniOpenFile", [message, files[0]])